lib/elixir/lib/code.ex

```diff
@@ -696,6 +696,11 @@ defmodule Code do
       `'foo'` becomes `~c"foo"`.
       Defaults to the value of the `:migrate` option. This option changes the AST.
 
+    * `:migrate_unless` (since v1.18.0) - when `true`,
+      rewrites `unless` expressions using `if` with a negated condition, for example
+      `if foo, do:` becomes `unless !foo, do:`.
+      Defaults to the value of the `:migrate` option. This option changes the AST.
```

Comment:
We do handle these, it becomes `if !(x = expr())` which is terrible code, but it does work fine and I expect `unless x = expr()` to be a very rare need anyway since `x` could only be `false` or `nil`. WDYT?

https://github.com/elixir-lang/elixir/pull/13769#issuecomment-2318273536
https://github.com/elixir-lang/elixir/pull/13769#issuecomment-2319063466